packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3150,6 +3150,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     ReplaceTextIntent: _replaceTextAction,
     UpdateSelectionIntent: _updateSelectionAction,
     DirectionalFocusIntent: DirectionalFocusAction.forTextField(),
+    DismissIntent: CallbackAction<DismissIntent>(onInvoke: (_) => hideToolbar(false)),
```

Comment:
I just tried this on a native iOS app on iPad with a physical keyboard, and pressing escape did nothing.  Maybe do a switch statement here and don't hide the toolbar on iOS?  And try to figure out what happens on Android if you can.

You can just merge this PR now if you want though since it's otherwise good to go.  Maybe open a separate PR to do the switch statement or open an issue for it.  Up to you.